compiler/rustc_lint/src/unused.rs

```diff
@@ -295,16 +405,33 @@ impl<'tcx> LateLintPass<'tcx> for UnusedResults {
             span: Span,
             descr_pre_path: &str,
             descr_post_path: &str,
+            attr_to_check: WhatAttrToCheck,
         ) -> bool {
+            let attr_name = match attr_to_check {
+                WhatAttrToCheck::MustUse => sym::must_use,
+                WhatAttrToCheck::MustBind => sym::must_bind,
+            };
             for attr in cx.tcx.get_attrs(def_id).iter() {
-                if cx.sess().check_name(attr, sym::must_use) {
-                    cx.struct_span_lint(UNUSED_MUST_USE, span, |lint| {
-                        let msg = format!(
-                            "unused {}`{}`{} that must be used",
-                            descr_pre_path,
-                            cx.tcx.def_path_str(def_id),
-                            descr_post_path
-                        );
+                if cx.sess().check_name(attr, attr_name) {
+                    let lint_type = match attr_to_check {
+                        WhatAttrToCheck::MustUse => UNUSED_MUST_USE,
+                        WhatAttrToCheck::MustBind => UNUSED_MUST_BIND,
+                    };
+                    cx.struct_span_lint(lint_type, span, |lint| {
+                        let msg = match attr_to_check {
+                            WhatAttrToCheck::MustUse => format!(
+                                "unused {}`{}`{} that must be used",
+                                descr_pre_path,
+                                cx.tcx.def_path_str(def_id),
+                                descr_post_path
+                            ),
+                            WhatAttrToCheck::MustBind => format!(
+                                "Bogus `let _ =` binding of {}`{}`{}.  Use something like `let _guard =`.",
```

Comment:
This should be changed to the same message suggested by [this comment](https://github.com/rust-lang/rust/pull/78715/files#r517053850).